### PR TITLE
Update: support .eslint.yaml as a valid config filename

### LINKED ISSUE #4045: Request: support .eslint.yaml as a valid config filename
The reason for this is simple: YAML syntax highlighting in editors.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/27165999-request-support-eslint-yaml-as-a-valid-config-filename?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by BYK**
Fixes #4045.

### CODE REVIEW

**Comment by BYK** [comment]
Gonna add tests and update docs if the approach looks good.

**Comment by nzakas** [comment]
Looks good!

File: lib/config.js [hunk 1]
```diff
@@ -24,16 +24,19 @@ var fs = require("fs"),
     isAbsolutePath = require("path-is-absolute"),
     isResolvable = require("is-resolvable"),
     validator = require("./config-validator"),
-    pathIsInside = require("path-is-inside");
+    pathIsInside = require("path-is-inside"),
+    find = require("lodash.find");
```
Comment (gyandeeps) [inline lib/config.js]: What is the advantage of using lodash find?

Comment (BYK) [inline lib/config.js]: I'd rather use the native `Array.prototype.find` but it is not available on all Node versions we support. If we don't use this, we'd have to implement the same logic ourselves, write tests for it and maintain it. I thought, "why bother when there's already something that did all that" :)

**Comment by BYK** [comment]
@nzakas - alright then. Gonna add tests and update the docs and ping here again.

File: lib/config.js (hunk 1 above)
Comment (gyandeeps) [inline lib/config.js]: I was just curious.

Comment (BYK) [inline lib/config.js]: Heh no it is okay. Sorry if I came off the wrong way. Was trying to explain what I was thinking in detail.

Comment (gyandeeps) [inline lib/config.js]: lol. All good man. :smile:

**Comment by BYK** [comment]
Updated: no longer a WIP.

**Comment by nzakas** [comment]
Looks like a good direction. Right now it looks like you so consider all config files found, but we should only accept one config file per directory. The config files with extensions should take priority based on alphabetical order (json, yaml, yml) over the extensionless version.

**Comment by nzakas** [comment]
@byk I'm going to take this over, hope you don't mind. :)

**Comment by BYK** [comment]
@nzakas totally fine. Sorry for the lack of movement. I hope to return back to my days of glory in December :(

**Comment by nzakas** [comment]
No worries!

**Comment by alberto** [comment]
Since the related issue is already solved, I think this branch should be deleted.

**Comment by nzakas** [comment]
Good point. @byk safe to delete?

**Comment by BYK** [comment]
Yup :)

**Pull request closed without merging**